### DISCUSSION TITLE
Modeling firebase subscriptions with selectors

### DISCUSSION THREAD

**Discussion by SunilChiluvuri** [facebookexperimental/Recoil#1863] [Q&A] [answered]
Hi, 

I'm currently using atom effects to model firebase subscriptions.
In this particular use case, based on user choice, the query for the subscription needs to change. 
As atom effects are not run again on a reset , there is no simple and clean way I see to pull this off currently. 
Is it possible to model data subscriptions using just a selector and use useRecoilRefresher to redo the subscription to a new query.

To simplify the question, is there something like setSelf of atom effects in selectors to asynchronously update the selector. 

Thanks

**Comment by drarmstr** [comment]
There isn't any callback to change the value once the evaluation has completed and been cached.  But, the setSelf() in the atom effect can be called asynchronously. So, you could subscribe to your user preference change in the Atom effect and then call setSelf(). Alternatively, you could use useRecoilCallback() to create a callback used when your preference changes to update the atom value.

**Comment by SunilChiluvuri** [comment]
Thanks for the quick response. By "you could subscribe to your user preference change in the Atom effect " do you mean using some mechanism in the recoil API or externally. Currently there is a settings atom that I use   "getLoadable" to read in the atom effect before setting up the firebase subscription. My understanding was that getLoadable is a one time read and not a subscription. Is there a way to subscribe to other atoms/selectors from within atom effects? Thanks

**Comment by drarmstr** [comment, marked as answer]
Yeah, I mean subscribing using whatever mechanism Firebase uses within the atom effect.  There is no way to subscribe to other atoms/selectors from an effect, though.  If you want to subscribe an effect to multiple atoms/selectors you can always use `useEffect()`.  That, coupled with `useRecoilCallback()` can enable you to subscribe to everything and update multiple atoms.
